test/release/examples/patterns/parallelIO.chpl

```diff
@@ -0,0 +1,171 @@
+    RangeChunk;  // break a range into chunks
+
+
+// declare some configuration constants for:
+//
+config const n = 100,                  // the array size per dimension
+             filename = "array.dat",   // the filename to use
+             numTasks = here.numPUs(), // the number of tasks for reading
+             printArray = true,        // whether to print the final array
+             debug = false;            // whether to print debugging output
+       
+
+// Create a copy of the locales array that arranges them in a
+// "vertical" arrangement so that each one gets a contiguous block
+// slab of elements (Chapel arrays are laid out in row-major order by
+// default).  This permits us to do one seek per locale (rather than a
+// seek per row) when using a block distribution, for simplicity in
+// this exercise.
+//
+const targetLocales = reshape(Locales, {1..numLocales, 1..1});
+
+
+// Create a distributed domain and array targeting our view of the
+// locales.
+//
+const D = {0..<n, 0..<n} dmapped Block({0..<n, 0..<n}, targetLocales);
+var A: [D] real;
+
+
+// Initialize the array with some distinctive values
+//
+forall (i,j) in D do
+  A[i,j] = i + j / n:real;
+
+
+// Create a new file for writing, and write the array's size:
+//
+var outfile = open(filename, mode=iomode.cw,
+                   style=new iostyle(binary=true));
+var outCh = outfile.writer();
+outCh.write(n);
+outCh.close();
+outfile.close();
+
+
+// Create a task per locale, on that locale, and have each one seek to
+// its appropriate spot in the file and write its local chunk of the
+// array.
+//
+coforall loc in targetLocales {
+  on loc {
+    // Which indices of A does my locale own?
+    //
+    const myIndices = D.localSubdomain();
+    const (myStartRow, myStartCol) = myIndices.low;
+    assert(myStartCol == 0);
+
+
+    // Where should I start writing in the file?
+    //
+    const skip = numBytes(int)            // the array size we wrote previously
+               + myStartRow * n * numBytes(real);     // the rows prior to ours
+
+
+    // Sanity check, if requested
+    //
+    if debug then
+      writeln("Locale ", here.id, " owns ", myIndices, " so skips ", skip);
+
+    // Get a reference to this locale's local block of data
+    //
+    ref myA = A.localSlice[myIndices];
+
+    // open the file again and an unlocked writing channel to it.
+    // Note that accessing a file on a remote locale currently doesn't
+    // work (this is an area where we don't currently paper over the
+    // fact that different locales are distinct processes).  This is
+    // why we're opening the file again here rather than simply
+    // referring to 'outfile' from above.
+    // 
+    var outCh = open(filename, mode=iomode.rw,
+                     style=new iostyle(binary=true)).writer(locking=false);
+    
+
+    // seek to the right spot in the file and write our local elements.
+    //
+    outCh.seek(skip);
```

Comment:
Since the seek is on a newly created channel, it would be better (as far as the I/O implementation is concerned) to pass the start position into the `.writer` call above (with the `start` argument). If it is not obvious to you that creating the channel and seeking it is equivalent to creating it with a different `start` then we need to adjust something in the docs here: https://chapel-lang.org/docs/master/modules/standard/IO.html#IO.file.writer

Since this is a tutorial, I would expect it to explain both. I have the same feedback for the reading part below.

Additionally, the I/O system is able to avoid some memory overhead in the case that the channel creation provides an `end` position. I think that should be possible in this case as well.